test/inductor/test_distributed_patterns.py

```diff
@@ -499,6 +527,209 @@ def test_fake_distributed_inductor(self):
         self._assert_same_grad(out1, out2)
 
 
+class FakeDistributedTests(torch._dynamo.test_case.TestCase):
+    def setUp(self):
+        super(
+            type(self), self
+        ).setUp()  # use explicit params for compiled autograd test wrapping
+        fake_store = FakeStore()
+        dist.init_process_group(
+            "fake", store=fake_store, rank=0, world_size=self.world_size
+        )
+
+    def tearDown(self):
+        super(
+            type(self), self
+        ).tearDown()  # use explicit params for compiled autograd test wrapping
+        dist.destroy_process_group()
+
+    @property
+    def device_type(self) -> str:
+        return "cuda"
+
+    @property
+    def world_size(self) -> int:
+        return 2
+
+    @skipIfRocm
+    @skipIfXpu
+    @requires_gpu()
+    @torch._functorch.config.patch(unsafe_allow_optimization_of_collectives=True)
+    def test_partitioner_saves_primal_instead_of_allgathered_primal(self):
+        import torch.distributed._functional_collectives as funcol
+
+        def scale(
+            t: torch.Tensor, amax_t: torch.Tensor
+        ) -> tuple[torch.Tensor, torch.Tensor]:
+            max_v = torch.finfo(torch.float8_e4m3fn).max
+            scale_t = torch.clamp(amax_t.float(), min=1e-12) / max_v
+            t_fp8 = (t / scale_t).to(torch.float8_e4m3fn)
+            return t_fp8, scale_t
+
+        def fp8_matmul(
+            first: torch.Tensor,
+            amax_first: torch.Tensor,
+            second_t: torch.Tensor,
+            amax_second_t: torch.Tensor,
+            parallel: str,
+        ) -> torch.Tensor:
+            tp_group = torch.distributed.group.WORLD
+            first_fp8, scale_first = scale(first, amax_first)
+            second_t_fp8, scale_second_t = scale(second_t, amax_second_t)
+
+            if parallel == "col":
+                first_fp8 = funcol.all_gather_tensor(
```

Comment:
related to the assumption about sharding below, iiuc input is 'data parallel' and this all-gather is to create a 'replicated' input?  Then I think it makes sense that we scale the whole input but only use our chunk/scale for Weight.